src/bartlett.jl

```diff
@@ -0,0 +1,69 @@
+# Tests for equality of covariance matrices
+
+export BartlettsTest
+
+abstract type CovarianceEqualityTest <: HypothesisTest end
+
+population_param_of_interest(::CovarianceEqualityTest) =
+    ("Equality of covariance matrices", NaN, NaN)
+
+## Utility functions
+
+# Finite population correction factor
+@inline _correction(p::Int, nxm1::Int, nym1::Int) =
+    1 - ((2p^2 + 3p - 1) * (inv(nxm1) + inv(nym1) - inv(nxm1 + nym1)) / 6(p + 1))
+
+# Pool the given covariance matrices, trashing them both
+function _poolcov!(Sx::AbstractMatrix, nxm1::Int, Sy::AbstractMatrix, nym1::Int)
+    Sx .*= nxm1
+    Sy .*= nym1
+    Sx .+= Sy
```

Comment:
If you compute `Sy` first, you can update `Sx` in a single line, which will allow for loop fusion.

Reply:
Very nice!

Reply:
Actually, this is the same function as in #145, so once that's merged I'll just remove this entirely, as the version in #145 is faster than what's here.